mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala

```diff
@@ -610,6 +639,38 @@ class LogisticRegressionSuite
     assert(model2.coefficients ~= coefficientsR relTol 1E-3)
   }
 
+  test("binary logistic regression with intercept without regularization with bound") {
+    val upperBoundOfCoefficients = Matrices.dense(1, 4, Array(1.0, 0.0, 1.0, 0.0))
+    val upperBoundOfIntercept = Vectors.dense(1.0)
```

Comment:
Can you check `setboundsOnIntercepts` and `.setFitIntercept(false)` should throw exception. Or one should override the other one.